core-bundle/src/Intl/Countries.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Intl;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\System;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Intl\Countries as SymfonyCountries;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class Countries
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var ContaoFramework
+     */
+    private $contaoFramework;
+
+    /**
+     * @var array<string>
+     */
+    private $countriesList;
+
+    public function __construct(TranslatorInterface $translator, RequestStack $requestStack, ContaoFramework $contaoFramework, array $countriesList)
+    {
+        $this->translator = $translator;
+        $this->requestStack = $requestStack;
+        $this->contaoFramework = $contaoFramework;
+        $this->countriesList = $countriesList;
+    }
+
+    /**
+     * @return array<string,string> Translated country names indexed by their uppercase ISO 3166-1 alpha-2 code
+     */
+    public function getCountries(string $displayLocale = null): array
+    {
+        if (null === $displayLocale && ($request = $this->requestStack->getCurrentRequest())) {
```

Comment:
```suggestion
        if (null === $displayLocale && null !== ($request = $this->requestStack->getCurrentRequest())) {
```
Maybe?

Reply:
I think ` && ($request = ` is the common coding style in the rest of the codebase (as long as there is no difference between null and a falsy value).

Reply:
Hmm, I find quite a few occurrences if I search for `null !== (` in the code base.

Examples for `$request`:

 https://github.com/contao/contao/blob/aaf1a3c4e809ff9bbace655505a326855691ede9/core-bundle/src/EventListener/InsertTags/DateListener.php#L151

https://github.com/contao/contao/blob/aaf1a3c4e809ff9bbace655505a326855691ede9/core-bundle/src/Framework/ScopeAwareTrait.php#L66

Reply:
Fixed in 608d942ae7fe3a25afc3a8afcf7e3645f9ed25f2